### PR TITLE
Fixes pancakes not being able to be made on the griddle

### LINKED ISSUE #56629: Pancake Batter not making pancakes on griddles.
Reporting client version: 513.1527

## Round ID:
[155633](https://scrubby.melonmesa.com/round/155633)

## Testmerges:
- [Dynamic 2021](https://www.github.com/tgstation/tgstation/pull/56221)
- [Tramstation: choo choo MORE MAP COMIN' THROUH](https://www.github.com/tgstation/tgstation/pull/56509)

## Reproduction:
I made pancake batter, poured out 5u into a beaker, and then splashed the griddle with the batter (harm intent). The 'raw' pancakes did not appear.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Puts the signal registration in initialization, and not in deconstruction. Fixes https://github.com/tgstation/tgstation/issues/56629

## Why It's Good For The Game

pancak

## Changelog
:cl:
fix: Fixes pancakes not being able to be made on the griddle.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 1]
```diff
@@ -35,7 +36,7 @@
 	if(default_deconstruction_crowbar(I, ignore_panel = TRUE))
 		return
 	variant = rand(1,3)
-	RegisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT, .proc/on_expose_reagent)
+	UnregisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT)
```
Comment (coiax) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:39]: Why do we need to unregister the signal... for when we're changing the variant of our griddle?

Comment (necromanceranne) [reply code/modules/food_and_drinks/kitchen_machinery/griddle.dm:39]: Nyoops. Okay Ill feex later tonight.

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 2]
```diff
@@ -27,6 +27,11 @@
 	grill_loop = new(list(src), FALSE)
 	if(isnum(variant))
 		variant = rand(1,3)
+	RegisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT, .proc/on_expose_reagent)
+
+/obj/machinery/griddle/Destroy()
+	UnregisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT)
+	. = ..()
```
Comment (Jared-Fogle) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:34]: Signals are unregistered when the object is destroyed, you shouldn't need this.

Comment (necromanceranne) [reply code/modules/food_and_drinks/kitchen_machinery/griddle.dm:34]: oh, well, great!

**Pull request merged**